### PR TITLE
Support insert by ActiveRecord

### PR DESCRIPTION
**Pull request by takeshit19**
I want to importing with ActiveRecord.
because save for encrypted column.

I thank you for reading it through.

### CODE REVIEW

File: lib/heart_seed/db_seed.rb [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
   module DbSeed
-    # delete all records and insert from seed yaml
+    # delete all records and bulk insert from seed yaml
     #
-    # @param source_file [String]
+    # @param file_path [String]
     # @param model_class [Class] require. extends {ActiveRecord::Base}
-    def self.bulk_insert(source_file: nil, model_class: nil)
-      fixtures = HeartSeed::Converter.read_fixture_yml(source_file)
+    def self.bulk_insert(file_path: nil, model_class: nil)
```
Comment (sue445) [inline lib/heart_seed/db_seed.rb]: Existing test is broken for this change. Please fix test too.

File: lib/heart_seed/db_seed.rb [hunk 2]
```diff
@@ -17,30 +17,49 @@ def self.bulk_insert(source_file: nil, model_class: nil)
       end
     end
 
+    # delete all records and insert from seed yaml
+    #
+    # @param file_path [String]
+    # @param model_class [Class] require. extends {ActiveRecord::Base}
+    def self.insert(file_path: nil, model_class: nil)
```
Comment (sue445) [inline lib/heart_seed/db_seed.rb]: Add test code for this method.

File: lib/heart_seed/db_seed.rb [hunk 3]
```diff
@@ -17,30 +17,49 @@ def self.bulk_insert(source_file: nil, model_class: nil)
...
+      Dir.glob(File.join(seed_dir, "*.yml")) do |file_path|
+        table_name = File.basename(file_path, '.*')
+        next unless target_table?(table_name, target_table_names)
 
-        ActiveRecord::Migration.say_with_time("#{file} -> #{table_name}") do
+        ActiveRecord::Migration.say_with_time("#{file_path} -> #{table_name}") do
           begin
             model_class = table_name.classify.constantize
-            bulk_insert(source_file: file, model_class: model_class)
+            if insert_mode == "bulk"
```
Comment (sue445) [inline lib/heart_seed/db_seed.rb]: Magic number `bulk` is scattered in this module ...
I think constant value (ex. `BULK`) is better than magic number.

File: lib/heart_seed/db_seed.rb [hunk 4]
```diff
@@ -17,30 +17,49 @@ def self.bulk_insert(source_file: nil, model_class: nil)
...
+    end
+
     # import all seed yaml to table
     #
-    # @param seed_dir [String]
-    # @param tables   [Array<String>,String] table names array or comma separated table names. if empty, import all seed yaml. if not empty, import only these tables.
-    # @param catalogs [Array<String>,String] catalogs names array or comma separated catalog names. if empty, import all seed yaml. if not empty, import only these tables in catalogs.
-    def self.import_all(seed_dir: HeartSeed::Helper.seed_dir, tables: ENV["TABLES"], catalogs: ENV["CATALOGS"])
+    # @param seed_dir    [String]
+    # @param tables      [Array<String>,String] table names array or comma separated table names. if empty, import all seed yaml. if not empty, import only these tables.
```
Comment (sue445) [inline lib/heart_seed/db_seed.rb]: Please properly newline (and others)

File: lib/heart_seed/db_seed.rb [hunk 5]
```diff
@@ -52,14 +80,21 @@ def self.import_all(seed_dir: HeartSeed::Helper.seed_dir, tables: ENV["TABLES"],
...
+    #                      if not empty, import only these tables.
+    # @param catalogs    [Array<String>,String] catalogs names array or comma separated catalog names.
+    #                      if empty, import all seed yaml.
+    #                      if not empty, import only these tables in catalogs.
+    # @param insert_mode [String] const ACTIVE_RECORD or other string.
+    #                      if empty or not ACTIVE_RECORD, using bulk insert.
+    #                      if ACTIVE_RECORD, import with ActiveRecord.
     # @param shard_names [Array<String>]
-    def self.import_all_with_shards(seed_dir: HeartSeed::Helper.seed_dir, tables: ENV["TABLES"], catalogs: ENV["CATALOGS"], shard_names: [])
+    def self.import_all_with_shards(seed_dir: HeartSeed::Helper.seed_dir, tables: ENV["TABLES"], catalogs: ENV["CATALOGS"], mode: ENV["MODE"] || BULK, shard_names: [])
```
Comment (cuonTK) [inline lib/heart_seed/db_seed.rb]: Use Newline on args? or use other way?

File: lib/heart_seed/db_seed.rb (hunk 1 above)
Comment (cuonTK) [inline lib/heart_seed/db_seed.rb]: Thanks. Fixed this.

File: lib/heart_seed/db_seed.rb (hunk 2 above)
Comment (cuonTK) [inline lib/heart_seed/db_seed.rb]: Thanks. I Added.
Please check spec.

**Comment by cuonTK** [comment]
Thank you for review!
And sorry to inexperienced.

complexity...orz

File: lib/heart_seed/db_seed.rb (hunk 5 above)
Comment (sue445) [inline lib/heart_seed/db_seed.rb]: newline 
(I often write newline in method signature)

File: lib/heart_seed/db_seed.rb [hunk 6]
```diff
@@ -17,30 +19,56 @@ def self.bulk_insert(source_file: nil, model_class: nil)
...
+    # @param tables      [Array<String>,String] table names array or comma separated table names.
+    #                      if empty, import all seed yaml.
+    #                      if not empty, import only these tables.
+    # @param catalogs    [Array<String>,String] catalogs names array or comma separated catalog names.
+    #                      if empty, import all seed yaml.
+    #                      if not empty, import only these tables in catalogs.
+    # @param insert_mode [String] const ACTIVE_RECORD or other string.
+    #                      if empty or not ACTIVE_RECORD, using bulk insert.
+    #                      if ACTIVE_RECORD, import with ActiveRecord.
+    def self.import_all(
```
Comment (sue445) [inline lib/heart_seed/db_seed.rb]: @cuonTK I'll fix after

File: lib/heart_seed/db_seed.rb (hunk 2 above)
Comment (sue445) [inline lib/heart_seed/db_seed.rb]: :100:

File: lib/heart_seed/db_seed.rb (hunk 6 above)
Comment (cuonTK) [inline lib/heart_seed/db_seed.rb]: Thanks!!

Comment (sue445) [inline lib/heart_seed/db_seed.rb]: @cuonTK I fixed !
https://github.com/takeshit19/heart_seed/pull/1

**Comment by sue445** [comment]
@cuonTK Released  `v0.0.4` !

**Comment by cuonTK** [comment]
Thanks!!

**Pull request merged**